src/lib/plot/hint.js

```diff
@@ -51,143 +84,250 @@ class Hint extends PureRenderComponent {
+    return align;
   }
 
   /**
    * Get a CSS mixin for a proper positioning of the element.
-   * @param {string} orientation Orientation.
+   * @param {Object} align object with horizontal and vertical prop strings.
    * @param {number} x X position.
    * @param {number} y Y position.
    * @returns {Object} Object, that may contain `left` or `right, `top` or
    * `bottom` properties.
    * @private
    */
-  _getOrientationStyle(orientation, x, y) {
-    let xCSS;
-    let yCSS;
-
-    if (orientation === ORIENTATION_BOTTOMLEFT ||
-      orientation === ORIENTATION_BOTTOMRIGHT) {
-      yCSS = this._getCSSTop(y);
-    } else {
-      yCSS = this._getCSSBottom(y);
-    }
-    if (orientation === ORIENTATION_TOPLEFT ||
-      orientation === ORIENTATION_BOTTOMLEFT) {
-      xCSS = this._getCSSRight(x);
-    } else {
-      xCSS = this._getCSSLeft(x);
-    }
-
+  _getAlignStyle(align, x, y) {
     return {
-      ...xCSS,
-      ...yCSS
+      ...this._getXCSS(align.horizontal, x),
+      ...this._getYCSS(align.vertical, y)
     };
   }
 
+  _getYCSS(verticalAlign, y) {
+    // obtain yCSS
+    switch (verticalAlign) {
+      case ALIGN.TOP_EDGE:
+        // this pins x to top edge
+        return this._getCSSTop(null);
+      case ALIGN.BOTTOM_EDGE:
+        // this pins x to bottom edge
+        return this._getCSSBottom(null);
+      case ALIGN.BOTTOM:
+        // this places hint text to the bottom of center, so set its top edge
+        return this._getCSSTop(y);
+      case ALIGN.TOP:
+      default:
+        // this places hint text to the top of center, so set its bottom edge
+        // default case should not be possible but if it happens set to BOTTOM
+        return this._getCSSBottom(y);
+    }
+  }
+
+  _getXCSS(horizontal, x) {
+    // obtain xCSS
+    switch (horizontal) {
+      case ALIGN.LEFT_EDGE:
+        // this pins x to left edge
+        return this._getCSSLeft(null);
+      case ALIGN.RIGHT_EDGE:
+        // this pins x to left edge
+        return this._getCSSRight(null);
+      case ALIGN.LEFT:
+        // this places hint text to the left of center, so set its right edge
+        return this._getCSSRight(x);
+      case ALIGN.RIGHT:
+      default:
+        // this places hint text to the right of center, so set its left edge
+        // default case should not be possible but if it happens set to RIGHT
+        return this._getCSSLeft(x);
+    }
+  }
+
   /**
-   * Get the class name from orientation value.
-   * @param {string} orientation Orientation.
-   * @returns {string} Class name.
+   * Get the class names from align values.
+   * @param {Object} align object with horizontal and vertical prop strings.
+   * @returns {string} Class names.
    * @private
    */
-  _getOrientationClassName(orientation) {
-    return `rv-hint--orientation-${orientation}`;
+  _getAlignClassNames(align) {
+    return `rv-hint--horizontalAlign-${align.horizontal}
+     rv-hint--verticalAlign-${align.vertical}`;
```

Comment:
We should leave in the `rv-hint--orientation-` classnames from before if the user is using orientation so that users that still depend on it can still use it.